XMonad/Actions/RotSlaves.hs

```diff
@@ -49,8 +52,8 @@ rotSlaves' :: ([a] -> [a]) -> Stack a -> Stack a
 rotSlaves' _ s@(Stack _ [] []) = s
 rotSlaves' f   (Stack t [] rs) = Stack t [] (f rs)                -- Master has focus
 rotSlaves' f s@(Stack _ ls _ ) = Stack t' (reverse revls') rs'    -- otherwise
-    where  (master:ws)     = integrate s
-           (revls',t':rs') = splitAt (length ls) (master:f ws)
+    where  (nonEmpty -> master :| ws)      = integrate s
+           (revls', nonEmpty -> t' :| rs') = splitAt (length ls) (master:f ws)
```

Comment:
`f` is an argument so it could be anything; again since this is exported it may be worth just returning the original stack on error here